Add application source, update Readme and Package.json

kyuuk opened the pull request:
- Added application source code
- Update gitignore for result directory
- Update Package.json with contributors field
- Update README.md

madarche commented:
This is a very good work. Some remarks:

* Rename `npm-repolint.js` into `src/index.js`
* Rename the `Module` directory in `src/lib`
* Add the line `"main": "src/index.js"` into the `package.json` file
* Write functions as functions `let existPackageJson = function(repo, file, done)` → `function existPackageJson(repo, file, done)`
* Make the test pass: `npm run test` must not return any error. Add `/* eslint-disable no-console */` at the top of the files that use the console legitimately
* Don't use globals

To not use globals you need to pass a `config` object to the `requester` and `tester` modules. Here is a way to do it:

`src/lib/requester.js`:
```javascript
let config
module.exports = function(a_config) {
    config = a_config
    return {
    searchRepo,
    search,
    get: simpleGet,
    getRaw: simpleGetRaw,
    setToken
    }
}
```

`src/index.js`:
```javascript
const config = {debug: false, root_dir_path: ''}
const checker = require('./Modules/tester')(config)
const requester = require('./Modules/requester')(config)
```
Do the following modifications and your PR will be accepted. Again, nice work.

kyuuk commented:
- Renamed npm-repolint.js into src/index.js
- Renamed the Module directory in src/lib
- Added "main": "src/index.js" in package.json
- updated functions as function name(...)
- lint test pass with no error
- deleted use of global

madarche commented:
@kyuuk 👍
